streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -63,83 +93,172 @@ public InternalTopicManager(final StreamsKafkaClient streamsKafkaClient,
+        final Set<InternalTopicConfig> topicsToBeCreated = validateTopicPartitions(topics.values(), existingTopicPartitions);
+        if (topicsToBeCreated.size() > 0) {
+            final Set<NewTopic> newTopics = new HashSet<>();
+
+            for (final InternalTopicConfig internalTopicConfig : topicsToBeCreated) {
+                final Properties topicProperties = internalTopicConfig.toProperties(windowChangeLogAdditionalRetention);
+                final Map<String, String> topicConfig = new HashMap<>(defaultTopicConfigs);
+                for (final String key : topicProperties.stringPropertyNames()) {
+                    topicConfig.put(key, topicProperties.getProperty(key));
+                }
+                newTopics.add(
+                    new NewTopic(
+                        internalTopicConfig.name(),
+                        internalTopicConfig.numberOfPartitions(),
+                        replicationFactor)
+                    .configs(topicConfig));
+            }
+
+            int remainingRetries = retries;
+            boolean retry;
+            do {
+                retry = false;
+
+                final CreateTopicsResult createTopicsResult = adminClient.createTopics(newTopics);
+
+                final Set<String> createTopicNames = new HashSet<>();
+                for (final Map.Entry<String, KafkaFuture<Void>> createTopicResult : createTopicsResult.values().entrySet()) {
+                    try {
+                        createTopicResult.getValue().get();
+                        createTopicNames.add(createTopicResult.getKey());
+                    } catch (final ExecutionException couldNotCreateTopic) {
+                        final Throwable cause = couldNotCreateTopic.getCause();
+                        final String topicName = createTopicResult.getKey();
+
+                        if (cause instanceof TimeoutException) {
+                            retry = true;
+                            log.debug("Could not get number of partitions for topic {} due to timeout. " +
+                                "Will try again (remaining retries {}).", topicName, remainingRetries - 1);
+                        } else if (cause instanceof TopicExistsException) {
+                            createTopicNames.add(createTopicResult.getKey());
+                            log.info(String.format("Topic %s exist already: %s",
+                                topicName,
+                                couldNotCreateTopic.getMessage()));
+                        } else {
+                            throw new StreamsException(String.format("Could not create topic %s.", topicName),
+                                couldNotCreateTopic);
+                        }
+                    } catch (final InterruptedException fatalException) {
+                        final String error = "Could not create internal topics.";
```

Comment:
nit: `Could not create internal topics because waiting on the response was interrupted; this should never happen.`

Also I'm thinking maybe we should just throw the `IllegalStateException` to be more specific?

Reply:
As above. Root exception is passed as parameters. Can change to `IllegalStateException`.